src/Build/Evaluation/Expander.cs

```diff
@@ -1478,12 +1478,25 @@ internal static string ConvertToString(object valueToConvert)
                 else
                 {
                     // The fall back is always to just convert to a string directly.
-                    convertedString = valueToConvert.ToString();
+                    // Issue: https://github.com/dotnet/msbuild/issues/9757
+                    if (IsNumberType(valueToConvert) && ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_10))
+                    {
+                        convertedString = Convert.ToString(valueToConvert, CultureInfo.InvariantCulture.NumberFormat);
+                    }
+                    else
+                    {
+                        convertedString = valueToConvert.ToString();
+                    }
                 }
 
                 return convertedString;
             }
 
+            private static bool IsNumberType(object obj)
+            {
+                return obj is double || obj is long || obj is int || obj is byte;
```

Comment:
Do we need to consider other numerical types (single, float, etc.)?

Reply:
Method removed, and the Convertion with invariant Culture enabled for any other type :)

Reply:
Update: The method is back with only double, long and int for now, to cover basic scenarios. Next iteration is to enable this for every type but on early preview next release.

Reply:
Sounds good - I support that! :-)